tests/test_compr.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief Set of tests to validate the different compression algorithms
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include <gtest/gtest.h>
+
+#include "always.h"
+#include "asciistring.h"
+#include "bufffileclass.h"
+#include "compressionmanager.h"
+#include "refpack.h"
+#include "zlibcompr.h"
+
+#define WRITE_DATA 0
+
+// Some special refpack mode that doesn't seem to be used by the game. Test it anyways :)
+TEST(compression, refpack_encode_quick)
+{
+    auto filepath = Utf8String(TESTDATA_PATH) + "/compr/uncompr.txt";
+    BufferedFileClass src_file(filepath.Str());
+    ASSERT_TRUE(src_file.Open(FM_READ));
+    size_t src_size = src_file.Size();
+    uint8_t *src_data = new uint8_t[src_size];
+    src_file.Read(src_data, src_size);
+
+    uint8_t *dst_data = new uint8_t[src_size + 8];
+    uint32_t dst_size = RefPack_Compress(dst_data, src_data, src_size, true);
+    EXPECT_GT(dst_size, 0);
```

Comment:
This compression result check looks quite generous, as it would allow error margin on compression result. We do expect compressed data to be always the same right? Perhaps hash contents of `dst_data` and compare hash value against expected hash value.

In multiple functions.

Reply:
It's true that these function don't really check a lot, but i wanted to avoid that different zlib library version may fail. E.g. it's valid that a newer zlib version compresses better than an older one

Reply:
Hmm, is it true that compression result can differ that much, but then can still be read back with any zlib implementation? I would have expected that results will remain identical with identical settings, as otherwise it would not be able to decompress it again. I could be wrong.

Reply:
Well, depending on the compression level the size can vary from 728 bytes (level 1) -> 656 bytes (level 9) with different sizes inbetween accordingly. When the zlib maintainer finds a way to further decrease the size while maintaining compability with the DEFLATE algorithm he will certainly implement it. Also it might be possible that the flags for the different levels change and the size may change.

I know this is very unlikely, but the library doesn't give any guarantees that compression ratio doesn't change with different versions. E.g. if we wanted to switch to `libdeflate`in the future (a better zlib pretty much) we could get even better ratios: https://github.com/ebiggers/libdeflate#compression-levels

Reply:
Is there a RefPack Decompress function? Perhaps pair that?

Reply:
What do you mean?

Reply:
Original data -> RefPack Compress -> RefPack Decompress -> Compare decompressed with original.

Reply:
I kind of do that with the current tests - i compress the files (and save them on disk) and then decompress those files on disk again. I don't compare the hashes, but atleast if the size matches.

However it's a good idea do make a roundtrip test, that does a `memcmp` on the decompressed data with the original

Reply:
Yes. I have seen the decompression in the other test. The simpler test would be to compress, compare, decompress, compare again, without writing compressed result to file. This reduces test file set. You would only need uncompr.txt. Or not even, you could make the text part of the cpp file and make everything in memory, skipping file IO entirely.